controller/src/main/java/org/jboss/as/controller/operations/common/XmlMarshallingHandler.java

```diff
@@ -2,103 +2,45 @@
  * Copyright The WildFly Authors
  * SPDX-License-Identifier: Apache-2.0
  */
-
 package org.jboss.as.controller.operations.common;
 
-import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.OP;
-import static org.jboss.as.controller.logging.ControllerLogger.MGMT_OP_LOGGER;
-
-import java.io.BufferedOutputStream;
 import java.io.ByteArrayOutputStream;
-import java.io.Closeable;
 import java.nio.charset.StandardCharsets;
-import java.util.Collections;
-import java.util.EnumSet;
-import java.util.Set;
+import org.jboss.as.controller.ModelVersion;
 
 import org.jboss.as.controller.OperationContext;
-import org.jboss.as.controller.OperationStepHandler;
-import org.jboss.as.controller.PathAddress;
 import org.jboss.as.controller.SimpleOperationDefinition;
 import org.jboss.as.controller.SimpleOperationDefinitionBuilder;
-import org.jboss.as.controller.access.Action;
-import org.jboss.as.controller.access.AuthorizationResult;
 import org.jboss.as.controller.access.management.SensitiveTargetAccessConstraintDefinition;
 import org.jboss.as.controller.descriptions.ModelDescriptionConstants;
 import org.jboss.as.controller.descriptions.common.ControllerResolver;
-import org.jboss.as.controller.logging.ControllerLogger;
 import org.jboss.as.controller.persistence.ConfigurationPersister;
-import org.jboss.as.controller.registry.Resource;
-import org.jboss.dmr.ModelNode;
 import org.jboss.dmr.ModelType;
 
 /**
  * A {@link org.jboss.as.controller.OperationStepHandler} that can output a model in XML form
  *
  * @author Brian Stansberry (c) 2011 Red Hat Inc.
  */
-public class XmlMarshallingHandler implements OperationStepHandler{
+public class XmlMarshallingHandler extends AbstractXmlMarshallingHandler {
 
     private static final String OPERATION_NAME = ModelDescriptionConstants.READ_CONFIG_AS_XML_OPERATION;
 
-    public static final SimpleOperationDefinition DEFINITION = new SimpleOperationDefinitionBuilder(OPERATION_NAME,ControllerResolver.getResolver())
+    public static final SimpleOperationDefinition DEFINITION = new SimpleOperationDefinitionBuilder(OPERATION_NAME, ControllerResolver.getResolver())
             .addAccessConstraint(SensitiveTargetAccessConstraintDefinition.READ_WHOLE_CONFIG)
             .setReplyType(ModelType.STRING)
+            .setDeprecated(ModelVersion.create(17))
```

Comment:
This lags behind the current model version for WF 31, which is 24.

Reply:
Since the replacement operation is in COMMUNITY stability only we shouldn't deprecate it

Reply:
@pferraro This seems to be something pending to address with the new feature streams, ability to conditionally add a deprecation

Reply:
Since the stability level of the server cannot be determined statically, you would need to define this during registration.
AIUI, you want to deprecate XmlFileMarshallingHandler.DEFINITION in favor of XmlMarshallingHandler.DEFINITION for Stability.COMMUNITY?
If so, you'll want to do something like:
e.g.
```java
resourceRegistration.registerOperationHandler(XmlFileMarshallingHandler.DEFINITION, new XmlFileMarshallingHandler(configurationPersister));
ModelVersion deprecation = resourceRegistration.enables(XmlFileMarshallingHandler.DEFINITION) ? ModelVersion.create(...) : null;
resourceRegistration.registerOperationHandler(XmlMarshallingHandler.createOperationDefinitionBuilder().setDeprecated(deprecation).build(), xmh);
```